src/TSHelper.ts

```diff
@@ -109,14 +109,31 @@ export class TSHelper {
             const comments = type.symbol.getDocumentationComment(checker);
             const decorators =
                 comments.filter(comment => comment.kind === "text")
-                        .map(comment => comment.text.trim().split("\n"))
+                        .map(comment => comment.text.split("\n"))
                         .reduce((a, b) => a.concat(b), [])
+                        .map(line => line.trim())
                         .filter(comment => comment[0] === "!");
+
             const decMap = new Map<DecoratorKind, Decorator>();
+
             decorators.forEach(decStr => {
-                const dec = new Decorator(decStr);
-                decMap.set(dec.kind, dec);
+                const dec = new Decorator(decStr.substr(1));
+                if (dec.kind !== undefined) {
```

Comment:
Looks a bit weird to create a decorator jsut to see if its valid, maybe we can move this check to a static function `Decorator::IsValid(string)`